ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/InferredTypeHintStage.fs

```diff
@@ -0,0 +1,63 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Stages
+
+open JetBrains.Application.Settings
+open JetBrains.ProjectModel
+open JetBrains.ReSharper.Daemon.Stages
+open JetBrains.ReSharper.Plugins.FSharp
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Settings
+open JetBrains.ReSharper.Psi
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Daemon.Cs.Stages
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open FSharp.Compiler.SourceCodeServices
+
+type InferredTypeHintHighlightingProcess(logger: ILogger, fsFile, settings: IContextBoundSettingsStore, daemonProcess: IDaemonProcess) =
+    inherit FSharpDaemonStageProcessBase(fsFile, daemonProcess)
+
+    override x.Execute(committer) =
+        let consumer = new FilteringHighlightingConsumer(daemonProcess.SourceFile, fsFile, settings)
+        fsFile.ProcessThisAndDescendants(Processor(x, consumer))
+        committer.Invoke(DaemonStageResult(consumer.Highlightings))
+
+    override x.VisitLocalReferencePat(localRefPat, consumer) =
+        match localRefPat.IgnoreParentParens().Parent with
+        | :? ITypedPat -> ()
+        | :? ILocalBinding as localBinding when localBinding.ReturnTypeInfo <> null -> ()
+        | _ ->
+
+        match localRefPat.Identifier.As<FSharpIdentifierToken>() with
```

Comment:
This is fine and is used in some places already, though, I'm trying to shape up some API for easier getting the identifier. Something like the following could work well too.
NameIdentifier may be compound here, like for active patterns or type parameters.
```fsharp
if isNull localRefPat.NameIdentifier.IdentifierToken then () else
```